frontend/src/app/pages/exercises/exercise/shared/simulation/signaller-modal/signaller-modal-interactions/signaller-modal-interactions.component.ts

```diff
@@ -0,0 +1,314 @@
+                            radiogram.informationRequestKey,
+                            this.clientId
+                        )
+                    )
+                ),
+                takeUntil(this.destroy$)
+            )
+            .subscribe((radiograms) => {
+                radiograms.forEach((radiogram) => {
+                    this.exerciseService.proposeAction({
+                        type: '[Radiogram] Accept radiogram',
+                        clientId: this.clientId,
+                        radiogramId: radiogram.id,
+                    });
+
+                    setLoadingState(
+                        this.interactions,
+                        getInformationRequestKeyDetails(
+                            radiogram.informationRequestKey!
+                        ),
+                        false
+                    );
+                });
+            });
+
+        this.requestedRadiograms$ = radiograms$.pipe(
+            map((radiograms) =>
+                radiograms.filter((radiogram) => isAccepted(radiogram))
+            ),
+            map((radiograms) =>
+                radiograms.filter(
+                    (radiogram) =>
+                        isInterfaceSignallerKeyForClient(
+                            radiogram.informationRequestKey,
+                            this.clientId
+                        ) &&
+                        radiogram.simulatedRegionId === this.simulatedRegionId
+                )
+            ),
+            map((radiograms) =>
+                groupBy(radiograms, (radiogram) =>
+                    getInformationRequestKeyDetails(
+                        radiogram.informationRequestKey!
+                    )
+                )
+            )
+        );
+    }
+
+    ngOnDestroy() {
+        if (this.hotkeyLayer) {
+            this.hotkeysService.removeLayer(this.hotkeyLayer);
+        }
+        if (this.filterLayer) {
+            this.hotkeysService.removeLayer(this.filterLayer);
+        }
+
+        this.destroy$.next();
+    }
+
+    onFilterFocus() {
+        this.filterActive = true;
+
+        this.filterLayer = this.hotkeysService.createLayer(true);
+        this.filterLayer.addHotkey(this.exitFilterHotkey);
+        this.filterLayer.addHotkey(this.upHotkey);
+        this.filterLayer.addHotkey(this.downHotkey);
+        this.filterLayer.addHotkey(this.confirmHotkey);
+        this.filterLayer.addHotkey(this.confirmSecondaryHotkey);
+    }
+
+    onFilterBlur() {
+        this.filterActive = false;
+        this.filter = '';
+        this.selectedIndex = -1;
+
+        if (this.filterLayer) {
+            this.hotkeysService.removeLayer(this.filterLayer);
+            this.filterLayer = null;
+        }
+    }
+
+    increaseSelectedIndex() {
+        if (this.selectedIndex + 1 < this.filteredInteractions.length)
+            this.selectedIndex++;
+    }
+
+    decreaseSelectedIndex() {
+        if (this.selectedIndex - 1 >= -1) this.selectedIndex--;
```

Comment:
```suggestion
        if (this.selectedIndex > 0) this.selectedIndex--;
```

Right now this seems to be incorrect as you allow `this.selectedIndex` to be `-1`. Or is this intended?

Reply:
This is on purpose. `selectedIndex == 0` indicates that the first option (index 0) is selected, while `selectedIndex == -1` indicates no selection.

In 1d63b5cb6bb0be1cdf76c37defa8a23f1b718605, I've rewritten the condition to be a bit easier to understand and added a comment to explain that the `>=` is not a typo.

Reply:
Ah, I see. Thanks for the explanation, I did not expect no selection being intended, but it's fine for me. Thanks for the updated condition and the comment, though.